Tests fail due to failing compilation of int_ops.c

<!--
If you're not sure whether what you're experiencing is a mypy bug, please see the "Question and Help" form instead.
Please consider:

- checking our common issues page: https://mypy.readthedocs.io/en/stable/common_issues.html
- searching our issue tracker: https://github.com/python/mypy/issues to see if it's already been reported
- asking on gitter chat: https://gitter.im/python/typing
-->

**Bug Report**

<!--
If you're reporting a problem with a specific library function, the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues

If the project you encountered the issue in is open source, please provide a link to the project.
-->

A lot of tests fail on FreeBSD with the following output:

```
_________________________________ testBuiltins _________________________________
data: /work/usr/ports/devel/py-mypy/work-py39/mypy-0.981/mypyc/test-data/run-misc.test:47:
/work/usr/ports/devel/py-mypy/work-py39/mypy-0.981/mypyc/test/test_run.py:144: in run_case
    self.run_case_inner(testcase)
/work/usr/ports/devel/py-mypy/work-py39/mypy-0.981/mypyc/test/test_run.py:160: in run_case_inner
    self.run_case_step(testcase, step)
/work/usr/ports/devel/py-mypy/work-py39/mypy-0.981/mypyc/test/test_run.py:272: in run_case_step
    assert False, "Compilation failed"
E   AssertionError: Compilation failed
E   assert False
----------------------------- Captured stdout call -----------------------------
running build_ext
building 'native' extension
creating build/temp.freebsd-13.1-RELEASE-amd64-cpython-39
creating build/temp.freebsd-13.1-RELEASE-amd64-cpython-39/build
cc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -fPIC -I/work/usr/ports/devel/py-mypy/work-py39/mypy-0.981/mypyc/lib-rt -I/usr/local/include/python3.9 -c build/__native.c -o build/temp.freebsd-13.1-RELEASE-amd64-cpython-39/build/__native.o -O0 -g1 -Werror -Wno-unused-function -Wno-unused-label -Wno-unreachable-code -Wno-unused-variable -Wno-unused-command-line-argument -Wno-unknown-warning-option
----------------------------- Captured stderr call -----------------------------
In file included from build/__native.c:4:
/work/usr/ports/devel/py-mypy/work-py39/mypy-0.981/mypyc/lib-rt/int_ops.c:547:30: error: shifting a negative signed value is undefined [-Werror,-Wshift-negative-value]
    if (y == -1 && x == -1LL << 63) {
                        ~~~~ ^
/work/usr/ports/devel/py-mypy/work-py39/mypy-0.981/mypyc/lib-rt/int_ops.c:565:30: error: shifting a negative signed value is undefined [-Werror,-Wshift-negative-value]
    if (y == -1 && x == -1LL << 63) {
                        ~~~~ ^
/work/usr/ports/devel/py-mypy/work-py39/mypy-0.981/mypyc/lib-rt/int_ops.c:610:30: error: shifting a negative signed value is undefined [-Werror,-Wshift-negative-value]
    if (y == -1 && x == -1LL << 31) {
                        ~~~~ ^
/work/usr/ports/devel/py-mypy/work-py39/mypy-0.981/mypyc/lib-rt/int_ops.c:628:30: error: shifting a negative signed value is undefined [-Werror,-Wshift-negative-value]
    if (y == -1 && x == -1LL << 31) {
                        ~~~~ ^
4 errors generated.
```

<!-- What went wrong? Paste mypy's output. -->

**Your Environment**

<!-- Include as many relevant details about the environment you experienced the bug in -->

- Mypy version used: 0.981
- Mypy command-line flags: -
- Mypy configuration options from `mypy.ini` (and other config files): -
- Python version used: 3.10.7
- FreeBSD 13.1
- Clang 13.0.0

--- a/mypyc/lib-rt/int_ops.c
+++ b/mypyc/lib-rt/int_ops.c
@@ -544,7 +544,7 @@ int64_t CPyInt64_Divide(int64_t x, int64_t y) {
         PyErr_SetString(PyExc_ZeroDivisionError, "integer division or modulo by zero");
         return CPY_LL_INT_ERROR;
     }
-    if (y == -1 && x == -1LL << 63) {
+    if (y == -1 && x == INT64_MIN) {
         PyErr_SetString(PyExc_OverflowError, "integer division overflow");
         return CPY_LL_INT_ERROR;
     }
@@ -562,7 +562,7 @@ int64_t CPyInt64_Remainder(int64_t x, int64_t y) {
         return CPY_LL_INT_ERROR;
     }
     // Edge case: avoid core dump
-    if (y == -1 && x == -1LL << 63) {
+    if (y == -1 && x == INT64_MIN) {
         return 0;
     }
     int64_t d = x % y;
@@ -607,7 +607,7 @@ int32_t CPyInt32_Divide(int32_t x, int32_t y) {
         PyErr_SetString(PyExc_ZeroDivisionError, "integer division or modulo by zero");
         return CPY_LL_INT_ERROR;
     }
-    if (y == -1 && x == -1LL << 31) {
+    if (y == -1 && x == INT32_MIN) {
         PyErr_SetString(PyExc_OverflowError, "integer division overflow");
         return CPY_LL_INT_ERROR;
     }
@@ -625,7 +625,7 @@ int32_t CPyInt32_Remainder(int32_t x, int32_t y) {
         return CPY_LL_INT_ERROR;
     }
     // Edge case: avoid core dump
-    if (y == -1 && x == -1LL << 31) {
+    if (y == -1 && x == INT32_MIN) {
         return 0;
     }
     int32_t d = x % y;

